cmd/kubeadm/app/util/etcd.go

```diff
@@ -49,3 +51,34 @@ func (cluster LocalEtcdCluster) GetEtcdClusterStatus() (*clientv3.StatusResponse
 
 	return resp, nil
 }
+
+// GetSecureEtcdClusterStatus returns nil for status Up or error for status Down
+func (cluster LocalEtcdCluster) GetSecureEtcdClusterStatus() (*clientv3.StatusResponse, error) {
```

Comment:
This new function is definitely necessary to add the L7 check to the upgrade DAG.
Will we need the non-secure one ever moving forward?
Might still be good to include in the lib.